chia/rpc/full_node_rpc_api.py

```diff
@@ -176,6 +178,36 @@ async def get_blockchain_state(self, _: Dict[str, Any]) -> EndpointResult:
             space = await self.get_network_space(
                 {"newer_block_header_hash": newer_block_hex, "older_block_header_hash": older_block_hex}
             )
+
+            try:
+                newer_block: Optional[BlockRecord] = peak
+                if newer_block is not None and newer_block.height > 0 and not newer_block.is_transaction_block:
+                    prev_hash = newer_block.prev_transaction_block_hash
+                    newer_block = self.service.blockchain.try_block_record(prev_hash) if prev_hash is not None else None
+                    if newer_block is None and prev_hash is not None:
+                        newer_block = await self.service.blockchain.get_block_record_from_db(prev_hash)
+
+                if newer_block is not None and newer_block.timestamp is not None:
+                    header_hash = self.service.blockchain.height_to_hash(uint32(max(1, newer_block.height - 4608)))
+                    older_block: Optional[BlockRecord] = (
+                        self.service.blockchain.try_block_record(header_hash) if header_hash is not None else None
+                    )
+                    if header_hash is not None and older_block is None:
+                        older_block = await self.service.blockchain.get_block_record_from_db(header_hash)
+
+                    if older_block is not None and older_block.height > 0 and not older_block.is_transaction_block:
+                        prev_hash = older_block.prev_transaction_block_hash
+                        if prev_hash:
```

Comment:
```suggestion
                        assert prev_hash is not None:
```

I'm pretty sure `prev_hash` is always `not None` as long as `older_block` is not the genesis block. I think it might be appropriate to put an `assert` for that here, instead of the check below

Reply:
I split the code around to seaprete free functin.